volttron/platform/control.py

```diff
@@ -856,6 +862,42 @@ def update_health_cache(opts):
         health_cache_timeout_date = datetime.now() + timedelta(seconds=health_cache_timeout)
 
 
+def modify_agent_rpc_authorizations(opts):
+    conn = opts.connection
+    agent_id = ".".join(opts.pattern[0].split(".")[:-1])
+    agent_method = opts.pattern[0].split(".")[-1]
+    if len(opts.pattern) < 2:
+        _log.error(f"Missing authorizations for method. "
+                   f"Should be in the format agent_id.method authorized_capability1 authorized_capability2 ...")
+        return
+    added_auths = [x for x in opts.pattern[1:]]
+    try:
+
+        auth_file = _get_auth_file(os.path.abspath(platform.get_home()))
+        entries = auth_file.read_allow_entries()
+        print(f'agent_id: {agent_id}\nEntries are:')
+        for entry in entries:
+            print(f"Entry: {entry}")
+            if entry.identity == agent_id:
+                if agent_method not in entry.rpc_method_authorizations:
+                    entry.rpc_method_authorizations[agent_method] = added_auths
+                elif not entry.rpc_method_authorizations[agent_method]:
+                    entry.rpc_method_authorizations[agent_method] = added_auths
+                else:
+                    entry.rpc_method_authorizations[agent_method].extend(
+                        [rpc_auth for rpc_auth in added_auths
+                         if rpc_auth in added_auths and
+                         rpc_auth not in entry.rpc_method_authorizations[agent_method]])
+                auth_file.update_by_index(entry, entries.index(entry))
+                return
+        _log.error(f"Agent identity not found in auth file!")
```

Comment:
Not sure if I'm reading this right, but this log error will be written every time this function is run. Is that the expected behavior?

Reply:
Good question, that is correct! If you take a look at the for loop before this error log entry, if the agent's identity is found within the auth file during the for loop, then the function will return, leaving the function before the log statement is written. However, if the agent's identity is not found in the auth file, then the for loop will run it's course without doing anything, and the error statement will be written to the log file.

Reply:
Ah, yes, you're right. I was reading too fast and didn't see `return`. FWIW, looking at the code more closely, would moving the log error to an if-statement that finds the targeted agent identity be more expressive?  For example, before entering the for loop, a check can be made up front (similar to a guard clause) such as the following (note: I might be iterating through the list of entries incorrectly, but you get the idea):

```python
entries = auth_file.read_allow_entries()

if agent_id not in set(e.identity for e in entries):
    _log.error(f"Agent identity not found in auth file!")
    return

print(f'agent_id: {agent_id}\nEntries are:')
for entry in entries:
    ....
return
```